Unity-Technologies/ScriptableRenderPipeline
# 2d/shader compatibility

### Purpose of this PR
Allows shaders created for 2D Renderer to show up when using the Forward Renderer, and shaders created for the Forward Renderer to show up when using the 2D Renderer.

---
### Release Notes
Fixed issue with certain LWRP shaders not showing when using forward/2D renderer.

---
### Testing status
https://katana.bf.unity3d.com/projects/com.unity.render-pipelines/builders?ScriptableRenderLoop_branch=2d%2Fshader-compatibility&unity_branch=trunk&automation-tools_branch=master&sort=1-asc

**Manual Tests**: What did you do?
- [Y] Opened test project + Run graphic tests locally
- [Y] Built a player
- [NA] Checked new UI names with UX convention
- [NA] Tested UI multi-edition + Undo/Redo
- [Y] C# and shader warnings (supress shader cache to see them)
- [NA] Checked new resources path for the reloader (in devloper mode, you have a button at end of resources that check the pathes)
- Other: 

**Automated Tests**: What did you setup? (Add a screenshot or the reference image of the test please)
I added two graphics tests 
069_2D_Forward_Shader_Compatibility_Forward
![069_2D_Forward_Shader_Compatibility_Forward](https://user-images.githubusercontent.com/47998819/58240401-80fda000-7d7d-11e9-86a6-ab621eb7e109.png)

070_2D_Forward_Shader_Compatibility_2D
![070_2D_Forward_Shader_Compatibility_2D](https://user-images.githubusercontent.com/47998819/58240454-95419d00-7d7d-11e9-9d5b-8e4c3da9f596.png)


Launch Katana (Link on master here - update for your branch):
https://katana.bf.unity3d.com/projects/com.unity.render-pipelines/builders?ScriptableRenderLoop_branch=2d%2Fshader-compatibility&unity_branch=trunk&automation-tools_branch=master&sort=1-asc

Alternative, launch Yamato (Select your branch):
https://yamato.prd.cds.internal.unity3d.com/jobs/78-ScriptableRenderPipeline/tree/2d%252Fshader-compatibility/.yamato%252Fupm-ci-lwrp.yml%2523All_LWRP/131426/job

Any test projects to go with this to help reviewers?
No

---
### Overall Product Risks
**Technical Risk**: None, Low, Medium, High?
Medium - While there were a lot of files changed, there were very few actual code changes.

**Halo Effect**: None, Low, Medium, High?
Low - Unlikely to affect systems beyond the ones touched.
---
### Comments to reviewers
Notes for the reviewers you have assigned.
